vent/menus/main.py

```diff
@@ -217,6 +217,33 @@ def perform_action(self, action):
                                      'action': form_action,
                                      'type': a_type,
                                      'cores': cores}}
+        # grammar rules
+        vowels = ['a', 'e', 'i', 'o', 'u']
+
+        # consonant-vowel-consonant ending
+        # Eg: stop -> stopping
+        if s_action[-1] not in vowels and \
+           s_action[-2] in vowels and \
+           s_action[-3] not in vowels:
+               form_args['action_dict'] = \
```

Comment:
most of this dict is repeated, why not create the dict first outside the if statements, and then only change the `present_t` key inside the if statements?  that will also probably help with your spacing issue.